### PR TITLE
Cmake Presets improvements

### PR DESCRIPTION
**Pull request by networkfusion**
## Description
* Simplifies and improves the CmakePresets
* The changes were (trying to be) individual so that they can be cherry picked.

## Motivation and Context
* Some of the options were obsolete/duplicated or un-necessary.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

### CODE REVIEW

**Comment by networkfusion** [comment]
!!! Ignoring code fix recommendations until the initial PR is checked over...

**Comment by networkfusion** [comment]
SWO rename removed as caused more issues than it is worth (for this PR).

**Comment by networkfusion** [comment]
Struggling with this one... 
image

All targets (as far as I can see) have the managed debugger enabled, but it is not the default option! Perhaps it should be changed, and a note about "RTM image..."

**Comment by networkfusion** [comment]
No further changes will be made against this PR, as I think it is enough (unless review comments are made).

**Comment by networkfusion** [comment]
No further changes will be made against this PR, as I think it is enough (unless review comments are made).

File: CMake/Modules/FindNF_NativeAssemblies.cmake [hunk 1]
```diff
@@ -161,12 +161,30 @@ endmacro()
 
 # Hardware.Esp32
 if(API_Hardware.Esp32)
-    ##### API name here (doted name)
+    if(API_Hardware.Stm32)
```
Comment (josesimoes) [inline CMake/Modules/FindNF_NativeAssemblies.cmake:164]: I understand what you're trying to do with this checks.
The thing is that the build will break ALWAYS if any of those is selected with the wrong platform. Which make these redundant.

With all these new IF statements it just gets harder to maintain and raises the need for updating them all when adding a new platform.

Having said that: I don't think we need these here.

Comment (networkfusion) [reply CMake/Modules/FindNF_NativeAssemblies.cmake]: Ok, I reverted this commit so the file was not changed.

**Review by josesimoes — APPROVED** [review]
LGTM

**Pull request merged**